### ISSUE TITLE
Invalid Count Workload Identity Submodule

### ISSUE THREAD

**Issue by M1Gd0nPG** [terraform-google-modules/terraform-google-kubernetes-engine#1631]
### TL;DR

I am getting an invalid count error when I attempt to use preexisting service accounts with the workload identity submodule.  I am able to successfully plan and deploy when I don't use preexisting service accounts.  I tried this on Terraform versions: 0.12 - 1.2 and none of them resolved the error

### Expected behavior

I expected a plan adding workload identity configurations and annotations to my pre-existing service accounts.

### Observed behavior

I received the following error:
` Error: Invalid count argument
│ 
│   on .terraform/modules/buildkite.kubernetes-engine_workload-identity.annotate-sa/main.tf line 57, in resource "random_id" "cache":
│   57:   count = (!local.skip_download) ? 1 : 0
│ 
│ The "count" value depends on resource attributes that cannot be determined
│ until apply, so Terraform cannot predict how many instances will be
│ created. To work around this, use the -target argument to first apply only
│ the resources that the count depends on.`

### Terraform Configuration

```hcl
# workload identity
# GCP service account to use
 resource "google_service_account" "preexisting" {
   account_id   = var.service_account_id
 }
 # K8 service account to use
 resource "kubernetes_service_account" "preexisting" {
   metadata {
     name      = var.k8_sa_name
     namespace = var.namespace
   }
 }
 module "kubernetes-engine_workload-identity" {
   source              = "terraform-google-modules/kubernetes-engine/google//modules/workload-identity"
   version             = "25.0.0"
   use_existing_gcp_sa = true
   use_existing_k8s_sa = true
   gcp_sa_name         = google_service_account.preexisting.account_id
   k8s_sa_name         = kubernetes_service_account.preexisting.metadata[0].name
   namespace           = kubernetes_service_account.preexisting.metadata[0].namespace
   project_id          = var.project
   name                = google_service_account.preexisting.account_id
   depends_on = [google_service_account.preexisting]
 }
(note: I tried various versions of "depends-on" as well.. adding the k8service account as an additional dependency, removing it altogether, etc... none of them worked)
```

### Terraform Version

```sh
I tried versions 0.12 to 1.2 (I went with highest patch version for each, so 1.2.9 is where I caved).
```

### Additional information

When I removed the pre-existing service account requirement, everything worked fine.  I would up just deleting the service account and having this module create it for me so I could move on, but I will hit this issue again as I try getting more services onboarded to workload identity.
